File: app/src/main/java/ru/vincetti/vimoney/data/adapters/AllCardsListRVAdapter.java
package ru.vincetti.vimoney.data.adapters;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.graphics.ColorUtils;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ru.vincetti.vimoney.R;
import ru.vincetti.vimoney.data.models.AccountListModel;

public class AllCardsListRVAdapter extends RecyclerView.Adapter<AllCardsListRVAdapter.CardsViewHolder> {
    private List<AccountListModel> data;
    private OnCardClickListener mListener;

    class CardsViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView accName, accType, accBalance, isArchive, accSymbol;
        View accContainer;

        CardsViewHolder(@NonNull View itemView) {
            super(itemView);
            accName = itemView.findViewById(R.id.acc_name);
            accType = itemView.findViewById(R.id.acc_type);
            accBalance = itemView.findViewById(R.id.acc_balance);
            isArchive = itemView.findViewById(R.id.acc_archive);
            accSymbol = itemView.findViewById(R.id.acc_symbol);
            accContainer = itemView.findViewById(R.id.acc_list_container);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            mListener.onCardClick(data.get(getAdapterPosition()).getId());
        }
    }

    public AllCardsListRVAdapter(OnCardClickListener listener) {
        this.mListener = listener;
    }

    @NonNull
    @Override
    public CardsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_all_cards_list
                , parent, false);
        return new CardsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CardsViewHolder holder, int position) {
        AccountListModel tmpAcc = data.get(position);
        holder.accName.setText(tmpAcc.getName());
        holder.accType.setText(tmpAcc.getType());
        holder.accBalance.setText(String.valueOf(tmpAcc.getSum()));
        holder.accSymbol.setText(tmpAcc.getSymbol());
        holder.accContainer.setBackgroundColor(ColorUtils.setAlphaComponent(Color.parseColor(tmpAcc.getColor()), 35));
        if(!tmpAcc.isArhive()){
            holder.isArchive.setVisibility(View.INVISIBLE);
        } else {
            holder.isArchive.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public int getItemCount() {
        if (data == null) {
            return 0;
        }
        return data.size();
    }

    public void setList(List<AccountListModel> accList) {
        data = accList;
        notifyDataSetChanged();
    }

    public interface OnCardClickListener {
        void onCardClick(int itemId);
    }
}

File: app/src/main/java/ru/vincetti/vimoney/data/adapters/AllCardsListRVAdapter.kt
package ru.vincetti.vimoney.data.adapters

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ru.vincetti.vimoney.R
import ru.vincetti.vimoney.data.models.AccountListModel

class AllCardsListRVAdapter(
        private val listener: (Int) -> Unit
) : RecyclerView.Adapter<AllCardsListRVAdapter.CardsViewHolder>() {
    private var data: List<AccountListModel>? = null

    inner class CardsViewHolder(
            itemView: View,
            val listener: (Int) -> Unit
    ) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        val accName: TextView = itemView.findViewById(R.id.acc_name)
        val accType: TextView = itemView.findViewById(R.id.acc_type)
        val accBalance: TextView = itemView.findViewById(R.id.acc_balance)
        val isArchive: TextView = itemView.findViewById(R.id.acc_archive)
        val accSymbol: TextView = itemView.findViewById(R.id.acc_symbol)
        val accLabel: View = itemView.findViewById(R.id.acc_label)

        init {
            itemView.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            data?.let {
                listener(it[adapterPosition].id)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CardsViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_all_cards_list, parent, false)
        return CardsViewHolder(view, listener)
    }

    override fun onBindViewHolder(holder: CardsViewHolder, position: Int) {
        data?.let {
            val tmpAcc = it[position]
            holder.accName.text = tmpAcc.name
            holder.accType.text = tmpAcc.type
            holder.accBalance.text = tmpAcc.sum.toString()
            holder.accSymbol.text = tmpAcc.curSymbol
            holder.accLabel.setBackgroundColor(Color.parseColor(tmpAcc.color))
            if (!tmpAcc.isArchive) {
                holder.isArchive.visibility = View.INVISIBLE
            } else {
                holder.isArchive.visibility = View.VISIBLE
            }
        }
    }

    override fun getItemCount(): Int {
        return data?.size ?: 0
    }

    fun setList(accList: List<AccountListModel>) {
        data = accList
        notifyDataSetChanged()
    }

}
